packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -2069,17 +2072,28 @@ export class Bucket extends BucketBase {
   }
 
   /**
-   * Allows the LogDelivery group to write, fails if ACL was set differently.
+   * Allows the 'logging.s3.amazonaws.com' service principal to write log events
+   * to this bucket.
    *
    * @see
-   * https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
+   * https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-server-access-logging.html
    */
-  private allowLogDelivery() {
-    if (this.accessControl && this.accessControl !== BucketAccessControl.LOG_DELIVERY_WRITE) {
-      throw new Error("Cannot enable log delivery to this bucket because the bucket's ACL has been set and can't be changed");
-    }
+  private allowLogDelivery(sourceBucket: IBucket, accountId: string, logFilePrefix?: string) {
+    const prefix = logFilePrefix ?? '';
 
-    this.accessControl = BucketAccessControl.LOG_DELIVERY_WRITE;
```

Comment:
Does this need to be added back then?

Reply:
Yes, I think so, which means that this change adds no value. Closing the PR until CloudFront (and possibly other services) catches up.